### PR TITLE
Show Google My Business tab only once connected

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/24015 that aims at:

* Enabling Google My Business back in the `development`, `wpcalypso`, and `stage` environments 
* Displaying the new stats tab only if a location has been connected
* Hiding the nudge on the `Stats` page if a location has been connected
* Grouping all Google My Business actions and selectors in the same place
* Removing some unused code

#### Testing instructions
 
1. Run `git checkout update/gmb` and start your server, or open a [live branch](https://calypso.live/?branch=update/gmb)
2. Log in to an account with a site eligible to Google My Business
3. Open the [`Stats` page](http://calypso.localhost:3000/stats)
4. Assert that no `Google My Business` tab is displayed in the header
5. Assert that the `Can your customers find you on Google?` nudge is displayed
6. Click the `Start Now` button on the nudge
7. Click the `Create Your Listing` button
8. Click the `Create Your Listing` button
9. Click the `Connect Location` button to access stats
10. Assert that a `Google My Business` tab is now displayed in the header
11. Click the `Traffic` tab
12. Assert that the nudge is no longer shown
13. Reload the page to restore initial state
14. Click the `X` icon in the upper right corner of the nudge
15. Assert that the nudge is no longer shown, even if you reload the page

#### Reviews
 
- [x] Code
- [x] Product

### CODE REVIEW

File: client/my-sites/google-my-business/select-location/index.js [hunk 1]
```diff
@@ -73,7 +78,7 @@ class GoogleMyBusinessSelectLocation extends Component {
 					<GoogleMyBusinessLocation isCompact key={ location.id } location={ location }>
 						<Button
 							href={ `/google-my-business/stats/${ siteSlug }` }
-							onClick={ this.trackConnectLocationClick }
+							onClick={ this.connectLocation.bind( null, location.id ) }
```
Comment (stephanethomas) [inline client/my-sites/google-my-business/select-location/index.js]: This is generating a linting error, but I couldn't find an elegant way to address it:

```javascript
81:8   error  JSX props should not use .bind()                                                      react/jsx-no-bind
81:18  error  JSX props should not use .bind()                                                      react/jsx-no-bind
```

**Comment by yurynix** [comment]
I followed the instructions and it works as described

File: config/stage.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
 		"external-media": true,
 		"external-media/google-photos": true,
 		"external-media/free-photo-library": true,
-		"google-my-business": false,
```
Comment (Tug) [inline config/stage.json]: stage should be very very close to production, I don't think it's a good idea to activate `google-my-business` there

File: client/state/preferences/schema.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export const remoteValuesSchema = {
 					type: 'object',
 					properties: {
 						dismissedAt: { type: 'number', minimum: 0 },
-						type: { type: 'string', enum: [ 'dismiss', 'already-listed' ] },
+						type: { type: 'string', enum: [ 'dismiss' ] },
```
[the remark this answers is not in the document]
Comment (stephanethomas) [reply client/state/preferences/schema.js]: I'm just removing code that we no longer use. As far as I know we dropped the option to let users specify that they already are all set on Google My Business, and don't need our assistance.

File: client/state/google-my-business/selectors.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+const getGoogleMyBusinessLocationId = ( state, siteId ) => {
+	return get( state, `googleMyBusiness.${ siteId }.location.id`, null );
+};
+
+export const isGoogleMyBusinessLocationConnected = ( state, siteId ) => {
```
[the remark this answers is not in the document]
Comment (stephanethomas) [reply client/state/google-my-business/selectors.js]: I'd like to keep `GoogleMyBusiness` in the first part of this selector name to act as namespace, something that will come in handy when scanning the `client/state/selectors` folder or using auto completion in IDEs.

File: client/blocks/google-my-business-stats-nudge/index.js [hunk 5]
```diff
@@ -20,8 +20,8 @@ import QueryPreferences from 'components/data/query-preferences';
 import {
 	isGoogleMyBusinessStatsNudgeDismissed,
 	getGoogleMyBusinessStatsNudgeDismissCount,
-} from 'state/selectors';
-import { dismissNudge } from './actions';
+} from 'state/google-my-business/nudge/selectors';
```
[the remark this answers is not in the document]
Comment (stephanethomas) [reply client/blocks/google-my-business-stats-nudge/index.js]: Alright, I'll move all selectors there.

File: client/my-sites/google-my-business/select-location/index.js [hunk 6]
```diff
@@ -70,14 +101,11 @@ class GoogleMyBusinessSelectLocation extends Component {
...
-							onClick={ this.trackConnectLocationClick }
-						>
-							{ translate( 'Connect Location' ) }
-						</Button>
-					</GoogleMyBusinessLocation>
+					<LocalizedGoogleMyBusinessLocationWithConnectButton
+						siteId={ siteId }
+						siteSlug={ siteSlug }
+						location={ location }
+					/>
```
Comment (stephanethomas) [inline client/my-sites/google-my-business/select-location/index.js]: This is still generating a warning for me:
```
Warning: Each child in an array or iterator should have a unique "key" prop.

Check the render method of `GoogleMyBusinessSelectLocation`. See https://fb.me/react-warning-keys for more information.
  in Localized(GoogleMyBusinessLocationWithConnectButton) (created by GoogleMyBusinessSelectLocation)
  in GoogleMyBusinessSelectLocation (created by Localized(GoogleMyBusinessSelectLocation))
  in Localized(GoogleMyBusinessSelectLocation) (created by Connect(Localized(GoogleMyBusinessSelectLocation)))
  in Connect(Localized(GoogleMyBusinessSelectLocation))
  in div (created by Layout)
  in div (created by Layout)
  in div (created by Layout)
  in Layout (created by Connect(Layout))
  in Connect(Layout) (created by ReduxWrappedLayout)
  in Provider (created by ReduxWrappedLayout)
  in ReduxWrappedLayout
```

File: client/my-sites/stats/site.jsx [hunk 7]
```diff
@@ -34,7 +34,7 @@ import { recordGoogleEvent } from 'state/analytics/actions';
 import PrivacyPolicyBanner from 'blocks/privacy-policy-banner';
 import ChecklistBanner from './checklist-banner';
 import GoogleMyBusinessStatsNudge from 'blocks/google-my-business-stats-nudge';
-import { isGoogleMyBusinessStatsNudgeVisible as isGoogleMyBusinessStatsNudgeVisibleSelector } from 'state/selectors';
+import { isGoogleMyBusinessStatsNudgeVisible as isGoogleMyBusinessStatsNudgeVisibleSelector } from 'state/google-my-business/nudge/selectors'; // eslint-disable-line max-len
```
[the remark this answers is not in the document]
Comment (stephanethomas) [reply client/my-sites/stats/site.jsx]: This is no longer a problem if we keep this selector where it was.

File: client/my-sites/google-my-business/select-location/index.js (hunk 6 above)
Comment (stephanethomas) [reply client/my-sites/google-my-business/select-location/index.js]: And then this error:

```
index.js:82 Uncaught TypeError: _this.props.connectGoogleMyBusinessLocation is not a function
  at GoogleMyBusinessLocationWithConnectButton._this.connectLocation (index.js:82)
  at HTMLUnknownElement.callCallback (react-dom.development.js:542)
  at Object.invokeGuardedCallbackDev (react-dom.development.js:581)
  at Object.invokeGuardedCallback (react-dom.development.js:438)
  at Object.invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:452)
  at executeDispatch (react-dom.development.js:836)
  at executeDispatchesInOrder (react-dom.development.js:858)
  at executeDispatchesAndRelease (react-dom.development.js:956)
  at executeDispatchesAndReleaseTopLevel (react-dom.development.js:967)
  at Array.forEach (<anonymous>)
```

I'm going to fix them.

**Comment by cbauerman** [comment]
Code + Changes look good, LGTM

**Pull request merged**